pkg/resources/custom_diffs.go

```diff
@@ -177,3 +180,22 @@ func ForceNewIfAllKeysAreNotSet(key string, keys ...string) schema.CustomizeDiff
 		return allUnset
 	})
 }
+
+func RecreateWhenUserTypeChangedExternally(userType sdk.UserType) schema.CustomizeDiffFunc {
+	return func(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
+		if n := diff.Get("user_type"); n != nil {
+			logging.DebugLogger.Printf("[DEBUG] new external value for user type %s\n", n.(string))
+			acceptableUserTypes := []string{string(userType)}
+			if userType == sdk.UserTypePerson {
```

Comment:
Nit: I'd move this logic to SDK.

Reply:
extracted as preinitialized `AcceptableUserTypes`